cephfs/admin/clone_test.go

```diff
@@ -143,3 +143,62 @@ func TestCloneSubVolumeSnapshot(t *testing.T) {
 		time.Sleep(5 * time.Millisecond)
 	}
 }
+
+func TestCancelClone(t *testing.T) {
+	fsa := getFSAdmin(t)
+	volume := "cephfs"
+	group := "Park"
+	subname := "Jurrasic"
+	snapname := "dinodna0"
+	clonename := "babydino2"
+
+	err := fsa.CreateSubVolumeGroup(volume, group, nil)
+	assert.NoError(t, err)
+	defer func() {
+		err := fsa.RemoveSubVolumeGroup(volume, group)
+		assert.NoError(t, err)
+	}()
+
+	svopts := &SubVolumeOptions{
+		Mode: 0750,
+		Size: 20 * gibiByte,
+	}
+	err = fsa.CreateSubVolume(volume, group, subname, svopts)
+	assert.NoError(t, err)
+	defer func() {
+		err := fsa.RemoveSubVolume(volume, group, subname)
+		assert.NoError(t, err)
+	}()
+
+	err = fsa.CreateSubVolumeSnapshot(volume, group, subname, snapname)
+	assert.NoError(t, err)
+	defer func() {
+		err := fsa.RemoveSubVolumeSnapshot(volume, group, subname, snapname)
+		assert.NoError(t, err)
+	}()
+
+	err = fsa.CloneSubVolumeSnapshot(
+		volume, group, subname, snapname, clonename,
+		&CloneOptions{TargetGroup: group})
+	var x NotProtectedError
+	if errors.As(err, &x) {
+		err = fsa.ProtectSubVolumeSnapshot(volume, group, subname, snapname)
+		assert.NoError(t, err)
+		defer func() {
+			err := fsa.UnprotectSubVolumeSnapshot(volume, group, subname, snapname)
+			assert.NoError(t, err)
+		}()
+
+		err = fsa.CloneSubVolumeSnapshot(
+			volume, group, subname, snapname, clonename,
+			&CloneOptions{TargetGroup: group})
+	}
+	assert.NoError(t, err)
+	defer func() {
+		err := fsa.ForceRemoveSubVolume(volume, group, clonename)
+		assert.NoError(t, err)
+	}()
+
+	err = fsa.CancelClone(volume, group, clonename)
```

Comment:
I wonder if this is racy. Cloning could be really quick, and if that is the case, cancel might not work?

Reply:
You are right, but in my practical tests on my laptop this hasn't failed and its worked in the CI so far too. That said, I'm not against improving it... just that it works for now. Since this is a test I'd rather try to improve it at a later time/PR

Reply:
Oh, it is more of something I was wondering about. Maybe a comment in the code helps to find inconsistent failures, in case it ever happens.